### PR TITLE
Drop stale tables ("history_*", "schema_revision", "synchro_version"), remove "version" columns from all the tables

### LINKED ISSUE #273: Remove stale tables
Remove `schema_revision` and `synchro_version` tables.

### LINKED ISSUE #280: Remove history tables
We agreed that these history tables are both hard to maintain and a potential performance issues (for migrations and mass update/delete). We will achieve the same objective with better event logs, sql query logs (provided by RDS) and point-in-time restoration (provided by RDS).
So let's remove all `history_*` tables, their triggers and the `version` attributes on all tables.

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #273 
Fixes #280

### CODE REVIEW

File: db/migrations/56_drop_stale_tables.sql [hunk 1]
```diff
@@ -0,0 +1,1467 @@
...
+-- +migrate StatementEnd
+DROP TRIGGER IF EXISTS `after_insert_filters`;
+DROP TRIGGER IF EXISTS `before_update_filters`;
+DROP TRIGGER IF EXISTS `before_delete_filters`;
+
+ALTER TABLE `groups` DROP INDEX `version`, DROP COLUMN `version`;
+DROP TABLE `history_groups`;
+DROP TRIGGER IF EXISTS `before_insert_groups`;
+-- +migrate StatementBegin
+CREATE TRIGGER `before_insert_groups` BEFORE INSERT ON `groups` FOR EACH ROW BEGIN IF (NEW.id IS NULL OR NEW.id = 0) THEN SET NEW.id = FLOOR(RAND() * 1000000000) + FLOOR(RAND() * 1000000000) * 1000000000; END IF; END
```
[the remark this answers is not in the document]
Comment (zenovich) [reply db/migrations/56_drop_stale_tables.sql]: Triggers and tests still use this feature creating new rows without explicitly set IDs. IMO, to get rid of it we need to remove ID columns from all the table where we don't really need them (like `groups_groups`, `groups_ancestors`, `groups_items` and so on).

[the remark this answers is not in the document]
Comment (smadbe) [reply db/migrations/56_drop_stale_tables.sql]: Ah yes the triggers. Maybe we could remove them except those which are required by the triggers (there are not so many creating rows).
For the tests, which ones for instance?

[the remark this answers is not in the document]
Comment (smadbe) [reply db/migrations/56_drop_stale_tables.sql]: (anyway do not change anything while I haven't confirmed they are not useful for something else)

[the remark this answers is not in the document]
Comment (zenovich) [reply db/migrations/56_drop_stale_tables.sql]: > For the tests, which ones for instance?

This one: https://github.com/France-ioi/AlgoreaBackend/blob/master/app/api/answers/submit.feature#L10
Actually there are lots of them.

[the remark this answers is not in the document]
Comment (smadbe) [reply db/migrations/56_drop_stale_tables.sql]: Mmh ok. You can leave it this way if you think it would cause too many troubles :-)

**Pull request merged**